x/evm/types/types.go

```diff
@@ -1883,3 +1884,27 @@ func GetMultisigAddresses(key tss.Key) ([]common.Address, uint8, error) {
 	threshold := uint8(key.GetMultisigKey().Threshold)
 	return KeysToAddresses(multisigPubKeys...), threshold, nil
 }
+
+// EventID ensures a correctly formatted event ID
+type EventID string
+
+// Validate returns an error, if the event ID is not in format of Hash-Index
+func (id EventID) Validate() error {
+	if err := utils.ValidateString(string(id)); err != nil {
+		return err
+	}
+
+	arr := strings.Split(string(id), "-")
```

Comment:
check that you get 2 parts after the split, otherwise the following might panic

Reply:
good catch